File: 159-b.py
def palindrome(str):
    return str == str[::-1]
str = input()
N = len(str)
former = int((N-1)/2)
letter = int((N+3)/2 - 1)
if palindrome(str) and palindrome(str[0:former]) and palindrome(str[letter:N]):
    print("Yes")
else:
    print("No")

File: 159-b.rb
def palindrome(str)
  str == str.reverse
end
str = gets.chomp!
N = str.length
if palindrome(str) and palindrome(str.slice(0..(N-1)/2)) and palindrome(str.slice((N+3)/2..N))
  print("Yes\n")
else
  print("No\n")
end
